### PR TITLE
created episode detail page

### PR DESCRIPTION
**Pull request by MarekTalpas**
This PR contains the implementation of the episode page.

### CODE REVIEW

File: src/components/episodeDescription/episodeDescription.css [hunk 1]
```diff
@@ -0,0 +1,29 @@
+.episode-description {
+  display: inline-block;
+  vertical-align: top;
+  width: calc(100% - 30% - 5vh);
```
Comment (dominik-git) [inline src/components/episodeDescription/episodeDescription.css]: I have never met with creating layout by using "calc" so I have no words and i have to approve it. Order browser like IE 9- do not support this technology so instead of "calc" you have to add  static value (width, height) in %.

Comment (MarekTalpas) [inline src/components/episodeDescription/episodeDescription.css]: Peter told me, that we do not support IE 9 and older versions, therefore the calc is acceptable to use.

Comment (varholak-peter) [inline src/components/episodeDescription/episodeDescription.css]: `vh` and `vw`are not supported, use percentage instead

File: src/components/episodeDescription/episodeDescription.css [hunk 2]
```diff
@@ -0,0 +1,29 @@
+.episode-description {
+  display: inline-block;
+  vertical-align: top;
```
Comment (varholak-peter) [inline src/components/episodeDescription/episodeDescription.css]: is this needed?

File: src/components/episodeDescription/episodeDescription.css (hunk 1 above)
Comment (varholak-peter) [inline src/components/episodeDescription/episodeDescription.css]: This applies to all other uses of `vh` and `vw`

File: src/components/episodeDescription/episodeDescription.css [hunk 3]
```diff
@@ -0,0 +1,29 @@
+.episode-description {
```
Comment (varholak-peter) [inline src/components/episodeDescription/episodeDescription.css]: This whole file is not BEM.

File: src/components/episodeDescription/episodeDescription.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// @flow
+import React from 'react';
+import './episodeDescription.css';
```
Comment (varholak-peter) [inline src/components/episodeDescription/episodeDescription.js]: we don't use camel case for file names, use `episode-description.css` instead

File: src/components/episodeDescription/episodeDescription.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+const EpisodeDescription = () => (
+  <div className="episode-description">
```
Comment (varholak-peter) [inline src/components/episodeDescription/episodeDescription.js]: BEM for every part again

File: src/components/episodeDescription/episodeDescription.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+// @flow
+import React from 'react';
+import './episodeDescription.css';
+
+const EpisodeDescription = () => (
```
Comment (varholak-peter) [inline src/components/episodeDescription/episodeDescription.js]: We need to pass all this data as props, currently this would create a component that would be same everywhere.

File: src/components/episodeDescription/episodeDescription.test.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+// @flow
+import React from 'react';
+import renderer from 'react-test-renderer';
+import EpisodeDescription from './episodeDescription';
+
+test('EpisodeDescription renders correctly', () => {
+  const tree : string = renderer.create(<EpisodeDescription />).toJSON();
```
Comment (varholak-peter) [inline src/components/episodeDescription/episodeDescription.test.js]: once you add information as props create tests according to it

File: src/components/episodeDetail/episodeDetail.css [hunk 8]
```diff
@@ -0,0 +1,9 @@
+.episode-detail {
```
Comment (varholak-peter) [inline src/components/episodeDetail/episodeDetail.css]: BEM again

File: src/components/episodeDescription/episodeDescription.js (hunk 4 above)
Comment (varholak-peter) [inline src/components/episodeDescription/episodeDescription.js]: This includes all other files.

File: src/components/episodeDetail/episodeDetail.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+// @flow
+import React from 'react';
+import EpisodeDescription from '../episodeDescription/episodeDescription';
+import EpisodeWatch from '../episodeWatch/episodeWatch';
+import EpisodeTiles from '../episodeTiles/episodeTiles';
+import './episodeDetail.css';
+
+const EpisodeDetail = () => (
```
Comment (varholak-peter) [inline src/components/episodeDetail/episodeDetail.js]: EpisodeDetail needs to be inside a container, which will fetch the data and pass it to EpisodeDetail, you can mock some fake data in container for now, but EpisodeDetail needs to use props.

File: src/components/episodeDetail/episodeDetail.test.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+// @flow
+import React from 'react';
+import renderer from 'react-test-renderer';
+import EpisodeDetail from './episodeDetail';
+
+test('EpisodeDetail renders correctly', () => {
+  const tree : string = renderer.create(<EpisodeDetail />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (varholak-peter) [inline src/components/episodeDetail/episodeDetail.test.js]: add empty line to end of file

File: src/components/episodeDetail/episodeDetail.test.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+// @flow
+import React from 'react';
+import renderer from 'react-test-renderer';
+import EpisodeDetail from './episodeDetail';
+
+test('EpisodeDetail renders correctly', () => {
+  const tree : string = renderer.create(<EpisodeDetail />).toJSON();
```
Comment (varholak-peter) [inline src/components/episodeDetail/episodeDetail.test.js]: again create test for various prop combinations

File: src/components/episodeTiles/episodeTile/episodeTile.css [hunk 12]
```diff
@@ -0,0 +1,13 @@
+.episode-tile {
+  display: inline-block;
+  width: 26%;
+  height: calc(100vh - 15vh - 40vh - 8vh - 8vh);
```
Comment (varholak-peter) [inline src/components/episodeTiles/episodeTile/episodeTile.css]: This is not good, others will not know what each of the number means and neither will you in 1 month.

File: src/components/episodeTiles/episodeTile/episodeTile.js [hunk 13]
```diff
@@ -0,0 +1,9 @@
+// @flow
+import React from 'react';
+import './episodeTile.css';
+
+const EpisodeTile = () => (
```
Comment (varholak-peter) [inline src/components/episodeTiles/episodeTile/episodeTile.js]: I would prefer "EpisodeCard" instead of "EpisodeTile" if you are okay with that, refactor accordingly.

File: src/components/episodeTiles/episodeTile/episodeTile.test.js [hunk 14]
```diff
@@ -0,0 +1,9 @@
+// @flow
+import React from 'react';
+import renderer from 'react-test-renderer';
+import EpisodeTile from './episodeTile';
+
+test('EpisodeTile renders correctly', () => {
+  const tree : string = renderer.create(<EpisodeTile />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (varholak-peter) [inline src/components/episodeTiles/episodeTile/episodeTile.test.js]: add empty line

File: src/components/episodeTiles/episodeTiles.js [hunk 15]
```diff
@@ -0,0 +1,16 @@
+// @flow
+import React from 'react';
+import EpisodeTile from './episodeTile/episodeTile';
+import './episodeTiles.css';
+
+const EpisodeTiles = () => (
```
Comment (varholak-peter) [inline src/components/episodeTiles/episodeTiles.js]: This will be a scrollbar in the end, you can remove EpisodeTiles, but keep the EpisodeTile component, which we will pass into the scrollbar.

File: src/components/episodeTiles/episodeTiles.test.js [hunk 16]
```diff
@@ -0,0 +1,9 @@
+// @flow
+import React from 'react';
+import renderer from 'react-test-renderer';
+import EpisodeTiles from './episodeTiles';
+
+test('EpisodeTiles renders correctly', () => {
+  const tree : string = renderer.create(<EpisodeTiles />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (varholak-peter) [inline src/components/episodeTiles/episodeTiles.test.js]: empty line

File: src/components/episodeWatch/episodeWatch.js [hunk 17]
```diff
@@ -0,0 +1,12 @@
+// @flow
+import React from 'react';
+import './episodeWatch.css';
+
+const EpisodeWatch = () => (
```
Comment (varholak-peter) [inline src/components/episodeWatch/episodeWatch.js]: This is a bad name, as it is not obvious what this component does.

Comment (varholak-peter) [inline src/components/episodeWatch/episodeWatch.js]: Again, we want to pass data as props here.

File: src/components/episodeWatch/episodeWatch.test.js [hunk 18]
```diff
@@ -0,0 +1,9 @@
+// @flow
+import React from 'react';
+import renderer from 'react-test-renderer';
+import EpisodeWatch from './episodeWatch';
+
+test('EpisodeWatch renders correctly', () => {
+  const tree : string = renderer.create(<EpisodeWatch />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (varholak-peter) [inline src/components/episodeWatch/episodeWatch.test.js]: empty line

File: src/components/episodeWatch/episodeWatch.test.js [hunk 19]
```diff
@@ -0,0 +1,9 @@
+// @flow
+import React from 'react';
+import renderer from 'react-test-renderer';
+import EpisodeWatch from './episodeWatch';
+
+test('EpisodeWatch renders correctly', () => {
+  const tree : string = renderer.create(<EpisodeWatch />).toJSON();
```
Comment (varholak-peter) [inline src/components/episodeWatch/episodeWatch.test.js]: add different tests based on various props.

File: src/index.css [hunk 20]
```diff
@@ -1,5 +1,6 @@
 body {
   margin: 0;
   padding: 0;
+  height: 100vh;
```
Comment (varholak-peter) [inline src/index.css]: Why is this needed?

File: src/components/episodeDescription/episodeDescription.css (hunk 1 above)
Comment (garySiladi) [inline src/components/episodeDescription/episodeDescription.css]: if we're talking about supporting Presto Opera, `calc` isn't supported either

File: src/components/episodeDescription/episodeDescription.css (hunk 2 above)
Comment (MarekTalpas) [inline src/components/episodeDescription/episodeDescription.css]: Yes, it is needed, without it watch and description components are scattered.
